### ISSUE TITLE
"Revise is not watching" pops up again

### ISSUE THREAD

**Issue by jaakkor2** [timholy/Revise.jl#431]
Very similar old issue is fine https://github.com/timholy/Revise.jl/issues/403 but if a line `using MAT` is included the warning `Revise is not watching` appears.
```julia
module debug_plotsrevise2

using Plots
using MAT
function __init__()
	pyplot()
end

end # module
```
```
julia> using debug_plotsrevise2
[ Info: Precompiling debug_plotsrevise2 [top-level]

julia> ┌ Warning: C:\Users\jaakkor2\.julia\packages\Plots\qZHsp\src\Plots.jl\ is not an existing directory, Revise is not watching
└ @ Revise C:\Users\jruohio\.julia\packages\Revise\ZOWOa\src\Revise.jl:489
```
Julia v1.3.1, Revise v2.5.1, MAT v0.7.0

Similar problem as in https://discourse.julialang.org/t/how-to-solve-warning-while-using-custom-package-with-revise/34743 .

**Comment by laborg** [comment]
The same happens to me if I use HDF5. A simple reproducer:
```
(@debughdf) pkg> st
Status `~/.julia/environments/debughdf/Project.toml`
  [f67ccb44] HDF5 v0.13.1
  [91a5bcdd] Plots v1.2.0

julia> using HDF5

julia> using Plots

julia> ┌ Warning: /home/gerhard/.julia/packages/Plots/EJmJT/src/Plots.jl/ is not an existing directory, Revise is not watching
└ @ Revise /home/gerhard/.julia/packages/Revise/MgvIv/src/Revise.jl:492
```

Using poor mans debugging methods I figured that somehow the `init_watching` gets called with files that are actually not part of the source of the package e.g. 
```
┌ Info: init_watching
│   files =
│    29-element Array{String,1}:
│     "/home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl"
│     "/home/gerhard/.julia/packages/Plots/EJmJT/src/init.jl__@require__"
│     "."
│     "/home/gerhard/.julia/packages/Plots/EJmJT/deps/deps.jl"
....
```
I couldn't find out where the `srcfiles` method gets tainted to contain the non-existing file `"src/init.jl__@require__"`  and  `"."`, which ultimately lead to the "Revise not watching message". I guess the culprit is somewhere within the combination of CodeTracking/Requires/Revise.

Plots `init.jl` calls `Revise.track()` if HDF5 is available in the `__init__` function via `@require`, but its  not clear how the workaround (`...sleep(0.1)...`) for  #403 should help here. I don't observe any differences if I deactivate this part of the code. 



A possible red herring I actually don't understand (why does it complain about the undefined N8f0?): 
```
.... # new Julia session
julia> using Plots

julia> Revise.track(Plots,"/home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl")
┌ Error: evaluation error starting at /home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl:46
│   mod = Plots
│   ex =
│    quote
│        #= toplevel:44 =#
│        if length(HDF5PLOT_MAP_TELEM2STR) < 1
│            #= /home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl:46 =#
│            telem2str = Dict{String, Type}("NATIVE" => HDF5PlotNative, "VOID" => Nothing, "BOOL" => Bool, "SYMBOL" => Symbol, "TUPLE" => Tuple, "CTUPLE" => HDF5CTuple, "RGBA" => ARGB{N0f8}, "EXTREMA" => Extrema, "LENGTH" => Length, "ARRAY" => Array, "DEFAULTSDICT" => DefaultsDict, "FONT" => Font, "BOUNDINGBOX" => BoundingBox, "GRIDLAYOUT" => GridLayout, "ROOTLAYOUT" => RootLayout, "SERIESANNOTATIONS" => SeriesAnnotations, "PLOTTEXT" => PlotText, "SHAPE" => Shape, "COLORGRADIENT" => ColorGradient, "AXIS" => Axis, "SURFACE" => Surface, "SUBPLOT" => Subplot, "NULLABLE" => (Union{Nothing, T} where T))
│            #= /home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl:73 =#
│            merge!(HDF5PLOT_MAP_STR2TELEM, telem2str)
│            #= /home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl:74 =#
│            merge!(HDF5PLOT_MAP_TELEM2STR, Dict{Type, String}((v => k for (k, v) = HDF5PLOT_MAP_STR2TELEM)))
│        end
│    end
│   exception =
│    UndefVarError: N0f8 not defined
│    Stacktrace:
│     [1] macro expansion at /home/gerhard/.julia/packages/JuliaInterpreter/dEBFI/src/interpret.jl:60 [inlined]
│     [2] getargs(::Array{Any,1}, ::JuliaInterpreter.Frame) at /home/gerhard/.julia/packages/JuliaInterpreter/dEBFI/src/builtins.jl:8
└ @ Revise ~/.julia/packages/Revise/MgvIv/src/lowered.jl:106
┌ Error: evaluation error starting at /home/gerhard/toplevel:288
│   mod = Plots
│   ex =
│    quote
│        #= toplevel:288 =#
│        function _hdf5plot_gwrite(grp, k::String, v::ARGB{N0f8})
│            #= /home/gerhard/.julia/packages/Plots/EJmJT/src/backends/hdf5.jl:289 =#
│            grp[k] = [v.r.i, v.g.i, v.b.i, v.alpha.i]
│            #= /home/
....
.... a couple more N0f8 not defined errors.
....
```

**Comment by timholy** [comment]
The N0f8 error is fixed for me, presumably by #486.

> I couldn't find out where the srcfiles method gets tainted to contain the non-existing file "src/init.jl__@require__" and ".", which ultimately lead to the "Revise not watching message".

That's done [here](https://github.com/timholy/Revise.jl/blob/26f14fe951631f4dd33bab83b26dc1d5e5e84dcb/src/pkgs.jl#L318-L327). The idea is that Revise needs the actual expression that Requires wants to run. Rather than make assumptions about what `@require` does, it [receives the expression from Requires](https://github.com/JuliaPackaging/Requires.jl/blob/master/src/require.jl#L82). Revise needs to associate that with a file, so it creates a fake file for all `@require` blocks. (It does something similar for methods defined at the REPL.)

@laborg, thanks for taking a close look at this! I'll try to get to this soon.
